src/components/LoginForm.tsx

```diff
@@ -0,0 +1,129 @@
+import styled from "@emotion/styled";
+import type { InputErrorHandlerHook } from "@src/hooks/useInputErrorHandler";
+import useInputErrorHandler from "@src/hooks/useInputErrorHandler";
+import useUserInfo, { type UserInfoHook } from "@src/hooks/useUserInfo";
+import theme from "@src/styles/kakaoTheme";
+import { createNewIDEvaluator } from "@src/utils/evaluator/implementation/idEvaluator";
+import { createNewPWEvaluator } from "@src/utils/evaluator/implementation/passwordEvaluator";
+import { useNavigate } from "react-router-dom";
+
+function LoginForm() {
+  const navigate = useNavigate();
+  const userInfo = useUserInfo();
+
+  const idEvaluator = createNewIDEvaluator();
+  const pwEvaluator = createNewPWEvaluator();
+  const inputErrorHandler: InputErrorHandlerHook = useInputErrorHandler();
+
+  const handleLogin = (userInfo: UserInfoHook) => {
+    console.log({
+      email: userInfo.email.value,
+      password: userInfo.password.value
+    });
+    if (window.history.length > 1) {
+      navigate(-1);
+    } else {
+      navigate("/");
+    }
+  };
+
+  return (
+    <InputForm onSubmit={(e) => e.preventDefault()}>
+      <h1>Kakao</h1>
+      <InputElement>
+        <InputField
+          type="text"
+          placeholder="이메일"
+          value={userInfo.email.value}
+          onChange={(e) => {
+            const newValue = e.target.value;
+            userInfo.email.setValue(newValue);
+            const isValid = idEvaluator.evaluate(newValue);
+            inputErrorHandler.idValid.setValue(isValid);
+            inputErrorHandler.idReason.setValue(idEvaluator.reason());
+          }}
+        />
+        {!inputErrorHandler.idValid.value && (
+          <ErrorP>{inputErrorHandler.idReason.value}</ErrorP>
+        )}
+      </InputElement>
+      <InputElement>
+        <InputField
+          type="password"
+          placeholder="비밀번호"
+          value={userInfo.password.value}
+          onChange={(e) => {
+            const newValue = e.target.value;
+            userInfo.password.setValue(newValue);
+            const isValid = pwEvaluator.evaluate(newValue);
+            inputErrorHandler.pwValid.setValue(isValid);
+            inputErrorHandler.pwReason.setValue(pwEvaluator.reason());
+          }}
+        />
+        {!inputErrorHandler.pwValid.value && (
+          <ErrorP>{inputErrorHandler.pwReason.value}</ErrorP>
+        )}
+      </InputElement>
+      <LoginButton
+        disabled={
+          !(
+            inputErrorHandler.idValid.value &&
+            inputErrorHandler.pwValid.value &&
+            userInfo.email.value.length > 0 &&
+            userInfo.password.value.length > 0
+          )
+        }
+        onClick={() => {
+          handleLogin(userInfo);
+        }}
```

Comment:
위 handleLogin을 수정한다면 여기도 이렇게 줄일 수 있겠죠?
```suggestion
        onClick={handleLogin}
```